lib/Sema/CSBindings.cpp

```diff
@@ -1631,8 +1640,22 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("subtype_of_existential");
   auto literalKind = getLiteralKind();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
+    std::string literalAttrStr;
+    if (literalKind == LiteralBindingKind::Atom) {
+      if (TypeVar->getImpl().getAtomicLiteralKind().hasValue()) {
+        auto atomKind = TypeVar->getImpl().getAtomicLiteralKind().getValue();
+        auto atomLiteral = TypeVar->getImpl().getAtomLiteral(atomKind).str();
+        literalAttrStr = ("[literal: " + atomLiteral + "]");
+      }
+    } else if (literalKind == LiteralBindingKind::Collection) {
+      for (const auto &protocolKind : getProtocolKind()) {
+        auto collectionLiteral = TypeVar->getImpl().getCollectionLiteral(protocolKind);
+        literalAttrStr = ("[literal: " + collectionLiteral + " " + getLiteralBindingKind(literalKind) + "]").str();
+      }
+    } else {
+      literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
+                              + "]").str();
+    }
     attributes.push_back(literalAttrStr);
```

Comment:
nit: this could be moved I think
```suggestion
    attributes.push_back(std::move(literalAttrStr));
```